src/NumberTheory/vinberg.jl

```diff
@@ -0,0 +1,291 @@
+Then also check the coorientation of `v` by checking that $v.v_0 \geq 0$.
+"""
+function _check_coorientation(Q::ZZMatrix, roots::Vector{ZZMatrix}, v::ZZMatrix, v0::ZZMatrix)
+  Qv = Q * transpose(v)
+  for r in roots
+    if (r*Qv)[1, 1] < 0
+      return false
+    end
+  end
+
+  return (v*Q*transpose(v0))[1, 1] > 0
+end
+
+@doc raw"""
+    _distance_0(Q::ZZMatrix, v0::ZZMatrix, root_lengths::Vector{ZZRingElem}) -> Vector{ZZMatrix}
+
+Return the roots which are orthogonal to `v0`.
+
+After gathering all primitive vectors `v` which are orthogonal to `v0` with length contained in `root_lengths`
+satisfying the crystallographic condition, we check for those `v` in order of increase of the value $\frac{(v.v_1)^2}{v^2}$ 
+if they are roots, e.g. if they also satisfy
+- $(v.v_1) > 0$
+- $(v.r) \geq 0$ for all roots `r` already found
+"""
+function _distance_0(Q::ZZMatrix, v0::ZZMatrix, root_lengths::Vector{ZZRingElem}, direction_vector::ZZMatrix)
+  roots = ZZMatrix[]
+  possible_vec = ZZMatrix[]
+  # _short_vectors_gram is more efficient than short_vectors_affine
+  # Thus we have to make some preparations
+  bm = kernel(Q*transpose(v0); side=:left)
+  QI = bm*Q*transpose(bm)
+  QI = QI[1, 1] > 0 ? QI : -QI
+  for d in root_lengths # gather all vectors which are orthogonal on v0 with length contained in root_lengths 
+    d = abs(d)
+    V = Hecke._short_vectors_gram(Hecke.LatEnumCtx, map_entries(QQ, QI), d, d, ZZRingElem) # QI POSITIVE
+    for (v_, _) in V
+      v = matrix(ZZ, 1, nrows(bm), v_) * bm
+      if !_is_primitive(v)
+        continue
+      end
+      if _crystallographic_condition(Q, v)
+        push!(possible_vec, v)
+      end
+    end
+  end
+  is_empty(possible_vec) && return roots
+  # we check if we can take the given direction vector or generate one if there is no one given
+  if iszero(direction_vector)
+    v1 = _generate_direction_vector(Q, v0, possible_vec)
+  else
+    v1 = _check_direction_vector(Q, v0, possible_vec, direction_vector)
+  end
+  # Now we want to sort the vectors by increase of the value ((v.v1)^2)//v^2
+  possible_vec_sort = Tuple{ZZMatrix,RationalUnion,RationalUnion}[]
+  for v in possible_vec
+    vQ = v * Q
+    n = (vQ*v1)[1, 1]
+    k = (vQ*transpose(v))[1, 1]
+    push!(possible_vec_sort, (v, n, k))
+  end
+  sort!(possible_vec_sort; by=x -> (x[2]^2) // abs(x[3]))
+
+  # We execute the algorithm with replacing v0 by v1
+  for (v, n) in possible_vec_sort
+    if n < 0
+      v = -v
+    end
+    vQ = v*Q
+    if all((vQ*transpose(w))[1, 1] >= 0 for w in roots)
+      push!(roots, v)
+    end
+  end
+  return roots
+end
+
+@doc raw"""
+    _generate_direction_vector(Q::ZZMatrix, v0::ZZMatrix, possible_vec::Vector{ZZMatrix}) -> ZZMatrix
+
+Since no direction vector `v1` was given, we need to find one which satisfies $(\~{v}, v_1) \neq 0$ 
+for all possible roots $\~{v}$ with $(v_0, \~{v}) = 0$
```

Comment:
```suggestion
Return a direction vector `v1`  satisfying $(\~{v}, v_1) \neq 0$ 
for all `possible_vec` $\~{v}$ with $(v_0, \~{v}) = 0$
```
Your docstring reads more like a comment in the code. But a docstring cannot assume that kind of context. 
Just state what the input is and what the output is. 
Further `v0` is not needed as input. You could remove that and compute `l` from `Q`.